### PR TITLE
[browser][mt] enable mono_gc_is_critical_method for MT build

### LINKED ISSUE #75970: [WASM] Check if we can still skip GC critical method check in WASM
There is an ifdef here to skip the critical method check in wasm: https://github.com/dotnet/runtime/blob/77bf509ab733114a04936cc6688c369f0dd2c090/src/mono/mono/metadata/sgen-mono.c#L237. 

Its not clear if this can still be safely skipped when wasm is multithreaded. Wasm uses co-op suspend, so *if* the managed code has no GC safepoints, this should still be ok. If it does have safepoints, the runtime might need to do the critical method check.

If we are able to safely skip it, the comment should be updated to say it can be skipped because of co-op suspend and the lack of safepoints (rather than single threadedness).

### PR DESCRIPTION
**Pull request by pavelsavara**
Fixes https://github.com/dotnet/runtime/issues/75970

### CODE REVIEW

**Comment by pavelsavara** [comment]
```
[18:27:32] info: Error: [MONO] * Assertion at /__w/1/s/src/mono/mono/sgen/sgen-gc.c:2849, condition `<disabled>' not met
[18:27:32] info: 
[18:27:32] info:     at ht (/root/helix/work/workitem/e/_framework/dotnet.runtime.js:3:12500)
[18:27:32] info:     at bt (/root/helix/work/workitem/e/_framework/dotnet.runtime.js:3:12754)
[18:27:32] info:     at wasm_trace_logger (wasm://wasm/00b16c8e:wasm-function[116]:0xa240)
[18:27:32] info:     at eglib_log_adapter (wasm://wasm/00b16c8e:wasm-function[884]:0x40f26)
[18:27:32] info:     at monoeg_g_logv_nofree (wasm://wasm/00b16c8e:wasm-function[802]:0x3d9f8)
[18:27:32] info:     at monoeg_assertion_message (wasm://wasm/00b16c8e:wasm-function[806]:0x3db19)
[18:27:32] info:     at mono_assertion_message (wasm://wasm/00b16c8e:wasm-function[808]:0x3db5c)
[18:27:32] info:     at mono_assertion_message_disabled (wasm://wasm/00b16c8e:wasm-function[807]:0x3db2f)
[18:27:32] info:     at sgen_gc_invoke_finalizers (wasm://wasm/00b16c8e:wasm-function[1104]:0x4ab70)
[18:27:32] info:     at mono_runtime_do_background_work (wasm://wasm/00b16c8e:wasm-function[3195]:0xdb091)
[18:27:32] info:     at mono_background_exec (wasm://wasm/00b16c8e:wasm-function[1024]:0x45cd1)
[18:27:32] info:     at Zo (/root/helix/work/workitem/e/_framework/dotnet.runtime.js:3:57330)
[18:27:32] info:     at callUserCallback (/root/helix/work/workitem/e/_framework/dotnet.native.js:8:112506)
[18:27:32] info:     at /root/helix/work/workitem/e/_framework/dotnet.native.js:8:112621
```
That is 
`g_assert (!pending_unqueued_finalizer);`

**Comment by vargaz** [comment]
That should be fixed already.

**Comment by pavelsavara** [comment]
I see https://github.com/dotnet/runtime/pull/89417 thanks

**Comment by pavelsavara** [comment]
Debugger CI failures are unrelated

File: src/mono/mono/metadata/sgen-mono.c [hunk 1]
```diff
@@ -232,7 +232,7 @@ sgen_has_critical_method (void)
 gboolean
 mono_gc_is_critical_method (MonoMethod *method)
 {
-#if defined(HOST_WASM) || defined(HOST_WASI)
+#if defined (DISABLE_THREADS) && (defined(HOST_WASM) || defined(HOST_WASI))
```
Comment (lewing) [inline src/mono/mono/metadata/sgen-mono.c:235]: Is DISABLE_THREADS alone not enough?

**Review by lewing — APPROVED** [review]

**Review by lambdageek — APPROVED** [review]

File: src/mono/mono/metadata/sgen-mono.c (hunk 1 above)
Comment (pavelsavara) [reply src/mono/mono/metadata/sgen-mono.c:235]: this is more descriptive anyway

**Pull request merged**